sdk/keyvault/Azure.Security.KeyVault.Administration/tests/AccessControlClientLiveTests.cs

```diff
@@ -32,6 +32,53 @@ public async Task GetRoleDefinitions()
             Assert.That(results[0].Type, Is.Not.Null);
         }
 
+        [RecordedTest]
+        public async Task UpsertRoleDefinition()
+        {
+            var description = Recording.GenerateAlphaNumericId("role");
+            var name = Recording.Random.NewGuid();
+            var originalPermissions = new KeyVaultPermission();
+            originalPermissions.DataActions.Add(KeyVaultDataAction.BackupHsmKeys);
+
+            KeyVaultRoleDefinition createdDefinition = await Client.UpsertRoleDefinitionAsync(description, originalPermissions, KeyVaultRoleScope.Global, name);
+
+            RegisterForCleanup(createdDefinition);
+
+            Assert.That(createdDefinition.AssignableScopes, Is.EqualTo(new[] { KeyVaultRoleScope.Global }));
+            Assert.That(createdDefinition.Description, Is.EqualTo(description));
+            Assert.That(createdDefinition.Name, Is.EqualTo(name.ToString()));
+            Assert.That(createdDefinition.Permissions.First().DataActions, Is.EquivalentTo(originalPermissions.DataActions));
+            Assert.That(createdDefinition.Type, Is.EqualTo(KeyVaultRoleDefinitionType.MicrosoftAuthorizationRoleDefinitions));
+
+            var updatedpermissions = new KeyVaultPermission();
+            updatedpermissions.DataActions.Add(KeyVaultDataAction.CreateHsmKey);
+            updatedpermissions.DataActions.Add(KeyVaultDataAction.DownloadHsmSecurityDomain);
+
+            KeyVaultRoleDefinition updatedDefinition = await Client.UpsertRoleDefinitionAsync(description, updatedpermissions, KeyVaultRoleScope.Global, name);
+
+            Assert.That(updatedDefinition.AssignableScopes, Is.EqualTo(new[] { KeyVaultRoleScope.Global }));
+            Assert.That(updatedDefinition.Description, Is.EqualTo(description));
+            Assert.That(updatedDefinition.Name, Is.EqualTo(name.ToString()));
+            Assert.That(updatedDefinition.Permissions.First().DataActions, Is.EquivalentTo(updatedpermissions.DataActions));
+            Assert.That(updatedDefinition.Type, Is.EqualTo(KeyVaultRoleDefinitionType.MicrosoftAuthorizationRoleDefinitions));
+        }
+
+        [RecordedTest]
+        public async Task DeleteRoleDefinition()
+        {
+            var description = Recording.GenerateAlphaNumericId("role");
+            var name = Recording.Random.NewGuid();
+            var originalPermissions = new KeyVaultPermission();
+            originalPermissions.DataActions.Add(KeyVaultDataAction.BackupHsmKeys);
+
+            KeyVaultRoleDefinition createdDefinition = await Client.UpsertRoleDefinitionAsync(description, originalPermissions, KeyVaultRoleScope.Global, name);
+            await Client.DeleteRoleDefinitionAsync(name, KeyVaultRoleScope.Global);
+
+            List<KeyVaultRoleDefinition> results = await Client.GetRoleDefinitionsAsync(KeyVaultRoleScope.Global).ToEnumerableAsync().ConfigureAwait(false);
```

Comment:
I noticed in passing that there's no `Client.GetRoleDefinition{Async}(scope, name)` and was wondering if that's something we want to purposely omit? I'm implementing it in JS so I figured it's worth asking (maybe I missed it). Thanks! 

https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.2-preview/rbac.json#L132

Reply:
Good catch  - I missed this operation!